File: February2022/Java/ExcelSheetColumnNumber.java

// @saorav21994
// TC = O(numberOfDigits)
// SC = O(1)  


class Solution {
    public int titleToNumber(String columnTitle) {
        char [] ch = columnTitle.toCharArray();
        int l = ch.length;
        int itr = 0, res = 0;
        for (int i = l-1; i >= 0; i--) {
            res += (int)Math.pow(26,itr)*(ch[i]-'A'+1);
            itr += 1;
        }
        return res;
    }
}

File: February2022/Kotlin/ExcelSheetColumnNumber.kt
//Author : Shobhit Behl (LC:shobhitbruh)
class Solution {
    fun titleToNumber(a: String): Int {
        val arr=a.toCharArray();
        var ans=0
        var p=0
        for(i in arr.indices.reversed()){
           ans+=(pow(p)*(arr[i]-'A'+1))
           p++
        }

        return ans
    }

    fun pow(x: Int):Int{
        var i=1
        var t=x
        while(t-->0){
            i*=26
        }

        return i
    }
}